stan/math/prim/meta/is_eigen_dense_plain_type.hpp

```diff
@@ -0,0 +1,33 @@
+#ifndef STAN_MATH_PRIM_META_IS_EIGEN_DENSE_PLAIN_TYPE_HPP
+#define STAN_MATH_PRIM_META_IS_EIGEN_DENSE_PLAIN_TYPE_HPP
+
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/prim/meta/bool_constant.hpp>
+#include <stan/math/prim/meta/require_helpers.hpp>
+#include <stan/math/prim/meta/is_eigen_dense_base.hpp>
+#include <stan/math/prim/meta/is_plain_type.hpp>
+#include <type_traits>
+
+namespace stan {
+
+/**
+ * Checks whether type T is derived from Eigen::DenseBase.
```

Comment:
```suggestion
 * Checks whether type T is a plain type derived from Eigen::DenseBase.
```

Reply:
Also, is this used anywhere?

Reply:
I think I used this in another version and can remove it now